pkg/composer/rm.go

```diff
@@ -92,3 +95,20 @@ func (c *Composer) removeContainers(ctx context.Context, containers []containerd
 
 	return nil
 }
+
+func (c *Composer) removeContainersFromParsedServices(ctx context.Context, containers map[string]serviceparser.Container) {
```

Comment:
Could not this entire logic be removed by calling `nerdctl run --rm`, rm should not be supported when detached is true.

Reply:
Do you mean calling `nerdctl run --rm` to do the job (run service container) when `nerdctl compose run --rm` is called? This function is directly moved from [here](https://github.com/containerd/nerdctl/blob/main/pkg/composer/run.go#L287C7-L300) which is called by `nerdctl compose run --rm` to process the `--rm` arg when needed.

This function is only a helper for deleting a list of containers (passed as `serviceparser.Container`) and doesn't do extra checks (similar to other compose helpers). The caller needs to do any checks it requires. For example, `compose run` checks `detach` [here](https://github.com/containerd/nerdctl/blob/main/pkg/composer/run.go#L263-L266) and returns directly. So this func won't be called when `detach` is true.

Reply:
Yep! Also can we keep in in run only as this is very specific to compose run.

Reply:
I think it's a good idea if it's doable (which I'm not sure :), because this func is after [`stopContainer`](https://github.com/containerd/nerdctl/blob/573d2a7bd661e527ebd69255bd2161da0f242593/pkg/composer/run.go#L284)). If it's doable to move `--rm` to `nerdctl run --rm` we should also take care of the `stopContainer` call above.

I'm okay with putting the function in either `composer/rm.go` or `composer/run.go`, but functionality-wise it's more related to `rm.go` so I put it there :). This func is actually not specific to `compose run` before. `compose up` also used [this function](https://github.com/containerd/nerdctl/blob/v1.0.0/pkg/composer/up_service.go#L89-L100) (which was a duplication) to `rm` containers when `stop` was not yet supported.

Reply:
Got it! cool! thanks for explaining.

Reply:
thanks for the review!